lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,313 @@
+import semver from "semver";
+const {SemVer: Version} = semver;
+import {promisify} from "node:util";
+import path from "node:path/posix";
+import {getLogger} from "@ui5/logger";
+const log = getLogger("builder:processors:manifestEnricher");
+
+const APP_DESCRIPTOR_V22 = new Version("1.21.0");
+
+/**
+ * Transforms i18n_en_US.properties to en_US
+ *
+ * @param {string} bundleName
+ * @param {string} defaultBundleName
+ */
+function getLocale(bundleName, defaultBundleName) {
+	log.verbose("Resource bundle detected on " + bundleName);
+	bundleName = path.basename(bundleName, ".properties");
+	defaultBundleName = path.basename(defaultBundleName, ".properties");
+	if (bundleName === defaultBundleName) {
+		return "";
+	} else {
+		return bundleName.replace(defaultBundleName + "_", "");
+	}
+}
+
+/**
+ * Determines if manifest contain handlebars placeholder templates
+ *
+ * Only in case of placeholders in manifest.json the bundle configured
+ * in sap.app/i18n is requested by sap/ui/core/Manifest.js
+ *
+ * @param {object} manifest The manifest
+ * @returns {boolean}
+ */
+function hasManifestTemplates(manifest) {
+	const rManifestTemplate = /\{\{([^}}]+)\}\}/g;
+	return JSON.stringify(manifest).match(rManifestTemplate);
+}
+
+/**
+ * Transforms "ui5://my/fancy/app/i18n/i18n.properties" to "my.fancy.app.i18n.i18n"
+ *
+ * @param {string} bundleUrl A bundle Url
+ * @returns {string}
+ */
+function getBundleNameByUrl(bundleUrl) {
+	return bundleUrl.replaceAll("ui5://", "").replaceAll("/", ".").replaceAll(".properties", "");
+}
+
+/**
+ * Determines if a bundle config should be enriched with supported locales
+ *
+ * @param {object} bundleConfig The bundle config
+ * @param {string} appId The app id
+ * @returns {boolean}
+ */
+function isAutofillRequired(bundleConfig, appId) {
+	const {supportedLocales, bundleUrl} = bundleConfig;
+	let {bundleName} = bundleConfig;
+
+	if (supportedLocales) {
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("/")) {
+		log.verbose("manifest.json: bundleUrl '" + bundleUrl +
+			"' contains an absolute path, no supportedLocales are generated");
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("..")) {
```

Comment:
So one could workaround this limitation by just specifying `./../../`, right?
I highly recommend to properly resolve the url and then check whether the path is within the expected namespace. In this case it's not a big problem, but in other scenarios such checks could lead to security vulnerabilities, so I'd recommend to always ensure proper handling of input.